alpha/action/generate_devfile.go

```diff
@@ -0,0 +1,53 @@
+package action
+
+import (
+	"fmt"
+	"io"
+	"text/template"
+)
+
+type GenerateDevfile struct {
+	IndexDir string
+	Writer   io.Writer
+}
+
+func (i GenerateDevfile) Run() error {
+	if err := i.validate(); err != nil {
+		return err
+	}
+
+	t, err := template.New("devfile").Parse(devfileTmpl)
+	if err != nil {
+		// The template is hardcoded in the binary, so if
+		// there is a parse error, it was a programmer error.
+		panic(err)
+	}
+	return t.Execute(i.Writer, i)
+}
+
+func (i GenerateDevfile) validate() error {
+	if i.IndexDir == "" {
+		return fmt.Errorf("index directory is unset")
+	}
+	return nil
+}
+
+const devfileTmpl = `schemaVersion: 2.2.0
+metadata:
+  name: fbc-4.11
```

Comment:
IMOs...

`name` should be a parameter, which in the composite case should be provided from components[].name
`displayName` should be a parameter, which in the composite case should be the same as `name`
`provider` should be a parameter since it's unbounded text and (I don't think) related to the processing logic connected to devfiles in RHTAP. 
`buildContext` should be a parameter, which in the composite case should be from catalogs.yaml `catalogs[].destination.workingDir`. 
`components[].image.ImageName` _may_ be desired to be parameterized, but since this is alpha and we're dry-fitting it I think it's probably fine not to expose it for now, and to iterate as necessary in future.